libfreerdp/core/multitransport.h

```diff
@@ -36,16 +36,22 @@ typedef enum
 	INITIATE_REQUEST_PROTOCOL_UDPFECL = 0x02
 } MultitransportRequestProtocol;
 
-FREERDP_LOCAL state_run_t multitransport_client_recv_request(rdpMultitransport* multitransport,
-                                                             wStream* s);
-FREERDP_LOCAL BOOL multitransport_server_send_request(rdpMultitransport* multitransport);
+typedef state_run_t (*MultiTransportRequestCb)(rdpMultitransport* multi, UINT32 reqId,
```

Comment:
Since `cookie` is not a string it might be a good idea to add a `cookieLen` argument as well (and not implicitly assume a certain size)

Reply:
Cookie is always 16 bytes, fixed size.

Reply:
no it can't change change it's hardcoded in the protocol, there's no way to have it to anything else

Reply:
or is it `\0` terminated?

Reply:
The cookie is just 16 bytes of random data that will have to be re-given on the UDP channel at the end of the negotiation, with the request id it make it match an incoming UDP connection with its TCP counterpart.